src/decoder/stream.rs

```diff
@@ -1074,6 +1128,33 @@ impl StreamingDecoder {
         Ok(State::ReadChunkData(type_str))
     }
 
+    fn record_ancillary_chunk(&mut self, stream_position: u64) {
+        if !self.decode_options.record_skipped_chunks {
+            return;
+        }
+
+        let chunk_start = stream_position - 8 - self.current_chunk.raw_bytes.len() as u64;
+
+        match self.current_chunk.type_ {
+            chunk::tEXt | chunk::zTXt | chunk::iTXt => {
+                if self.limits.reserve_bytes(8).is_ok() {
+                    self.text_chunk_positions.push(chunk_start);
+                }
+            }
+            chunk::eXIf => {
+                if self.exif_position.is_none() {
+                    self.exif_position = Some(chunk_start);
+                }
```

Comment:
The error treatment here differs from parsing the `eXIf` and `iCCP` chunk. In the latter case we error first with a `Format` and then later, probably, downgrade that to `BadAncillaryChunk` in most cases. Here however we do nothing.